### PR TITLE
Fine tune and compact more quick access panel

### PR DESCRIPTION
**Pull request by Nilvus**
As discussed on #8237

See before/after:

 

@elstoc, @AlicVB, @todd-prior and @TurboGit: hope it's good for you about compactness (not too much and not too less!) and how it's rendered now.

### CODE REVIEW

**Comment by elstoc** [comment]
I like the compactness. I'm still not sure whether I prefer the line above or below. An alternative might be to use shading and make it look closer to the actual header/body of the full modules.

**Comment by Nilvus** [comment]
> I like the compactness. I'm still not sure whether I prefer the line above or below (though below is more consistent with the rest of the UI). An alternative might be to use shading and make it look closer to the actual header/body of the full modules.

I see no problem of consistency here as quick access panel is just different like other parts. We do not have modules inside, like on all other parts. I find top border better as this just better separate lines related to a module. Anyway, with your shade idea, maybe something better could be proposed.

I set that PR in WIP so and will test that later. Just too late for this day. Thanks for your reply.

**Comment by elstoc** [comment]
> These are obviously the wrong colours...

Actually thinking about it, reversing the standard shading of the normal modules might be a good visual indication that you're in the quick access panel.

**Comment by todd-prior** [comment]
Could these be collapsed? So clicking on the shaded bar would toggle collapsed and expanded. If you added a toggle to open all and close all you could clean it up. You could offer open all or close all as a starting option.You could manage the look and keep things compact no matter how you construct the widget ie no matter what gets added to it??

**Comment by TurboGit** [comment]
Yes, I love @elstoc proposal even more!

**Comment by AlicVB** [comment]
I think @todd-prior raise an interesting point : at first, I quite like @elstoc proposal, but the main drawback is that it may imply in user mind that we can collapse things under the module header.
But that's not the case, and I really would prefer that we don't add such "features". Otherwise, quick access will quickly became a module ersatz at least for average users.
As you know, I'm a bit afraid that "quick panel" slowly became a monster out of control that will be used to deviate from dt workflow...

**Comment by TurboGit** [comment]
> As you know, I'm a bit afraid that "quick panel" slowly became a monster out of control that will be used to deviate from dt workflow...

I understand you, we must really refrain from adding too many GUI things in there.

**Comment by elstoc** [comment]
Yes we need to make it clear that it's not the same as the normal module UI (and not try to duplicate too much of its functionality), but I'm not sure I like the lines in the current proposal. Perhaps if the lines went the entire width of the panel and were a bit thicker.

Something like this...

Screenshot_2021-02-17_19-19-03

**Comment by Nilvus** [comment]
I will finally do exactly same answer as @AlicVB. I have just tested @elstoc proposal and indeed it let's think about modules way, so the idea to collapse/expand them. And that's not the idea of quick access panel.

So I will not add anything else in CSS. I think we should keep it as it is (if compactness is good for all of you) and keep in mind what this part is: a quick access panel (and not a whole bunch of reduced modules).

By writing that, I just see new @elstoc proposal and it's indeed much much better. I will work on that.

**Comment by Nilvus** [comment]
Last commit add @elstoc proposal with little finetune. Should be good now to review and merge (if ok for all).

**Comment by elstoc** [comment]
@Nilvus please can you add the following to make the on/off and full module link buttons the same size as the main UI
```
#basics-box-labels #module-enable-button, #basics-link
{
  min-height: 1.15em;
  min-width: 1.15em;
}
```

**Comment by Nilvus** [comment]
> @Nilvus please can you add the following to make the on/off and full module link buttons the same size as the main UI
> 
> ```
> #basics-box-labels #module-enable-button, #basics-link
> {
>   min-height: 1.15em;
>   min-width: 1.15em;
> }
> ```

I have reduce them to make them same size as text. In main UI, label text (module name) is bigger. And quick access panel is not modules so I disagree to have them same size (and same size would be 1em, not 1.15em). It's consistent as it's not same view/way and I made on/off size related to module name text, like in main UI.

**Comment by elstoc** [comment]
Sorry it looks too small with my font size - I use 8.4 font and it looks odd

Screenshot_2021-02-17_20-31-11

> (and same size would be 1em, not 1.15em)

I took the sizes from darktable.css

**Comment by Nilvus** [comment]
> Sorry it looks too small with my font size - I use 8.4 font and it looks odd

I use and test with a more standard font size and it doesn't look odd (see screenshot above) but I agree with your font size, that looks odd. Anyway, 1.15em or even 1em is also odd for me with more classic font size (11 or 12px).

**Comment by Nilvus** [comment]
Finally, with more test on my side, Height icon is consistent with height module name text  and not odd for me. But setting 1em make icons too big regardless of module name text (which is smaller in comparison of classic modules names in other groups). Setting 1.15em is worst and really awful for me. I don't see good compromise here but 0.85em remains for me the only consistent and finetune one regardless of module name text.

**Comment by elstoc** [comment]
ok but could we make just it a bit bigger than the text (the module ones are a bit bigger than the header text so it would be consistent) - perhaps 0.9 or 0.95

**Comment by todd-prior** [comment]
You’ll be revisiting this several times over no doubt. I think the cat is out of the bag when you allow people to pick from all the modules. You will have a monster.

This offers amazing flexibility and people with try to leverage it as much as they can leading to requests to be able to do this and that…..

I think you will have many people wanting a custom dashboard to cherry pick modules…Almost like a customized selection of their favourites packaged in a dashboard.

So I think you will have a monster… 😊

It’s a big endeavour.  Thanks for all the work on this….

**Comment by elstoc** [comment]
> I think the cat is out of the bag when you allow people to pick from all the modules. You will have a monster.

Maybe. The way I look at it though is it's just an alternative to keyboard shortcuts and it's no longer a way to bypass anything (like the old basic adjustments module was) because it's just another way to interact with the proper module controls.

**Comment by todd-prior** [comment]
I agree for sure.

**Comment by Nilvus** [comment]
> ok but could we make just it a bit bigger than the text (the normal module ones are a bit bigger than the header text so it would be consistent) - perhaps 0.9 or 0.95

Should be a good compromise. Update done and all should be good now. @TurboGit: ready for review and merge!

> You’ll be revisiting this several times over no doubt. I think the cat is out of the bag when you allow people to pick from all the modules. You will have a monster. This offers amazing flexibility and people with try to leverage it as much as they can leading to requests to be able to do this and that…..

You are right and we will need to be there to limit that and avoid having a monster by refraining requests and explain what it is and why we need to keep what it is for what it is and not a monster!

File: data/themes/darktable.css [hunk 1]
```diff
@@ -1527,35 +1527,25 @@ margin: 0 2px;
...
-}
-
+#basics-box-labels #module-enable-button,
 #basics-link
 {
-  min-height: 1em;
-  min-width: 1em;
-  margin: 0 0 0.3em 0.75em;
+  min-height: 0.95em;
+  min-width: 0.59em;
```
Comment (elstoc) [inline data/themes/darktable.css:1541]: whoops - typo?

Comment (Nilvus) [reply data/themes/darktable.css]: Just see that and already fixed! Thanks!

**Comment by elstoc** [comment]
All good for me now. Thanks @Nilvus.

**Comment by TurboGit** [comment]
Thanks !

**Comment by dterrahe** [comment]
Very small suggestion. I think the buttons on left and right of the header look ever so slightly better when hovered over if they have valign set to GTK_ALIGN_CENTER (rather than GTK_ALIGN_FILL as now)
imageimage

**Comment by elstoc** [comment]
@dterrahe agreed. More consistent with the full module UI.

**Comment by Nilvus** [comment]
@dterrahe: I agree too. As it's Gtk code and I don't know anything about Gtk code, could you do it?

**Comment by elstoc** [comment]
I'm on it

**Pull request merged**